### PR TITLE
welcoming robots in certain situations

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change

Right now, the same robots.txt file is served across all environments, in which all indexing is allowed, leading to Google indexing non-prod environments. This PR makes the robots.txt file disallow all by default and adds a bash script that runs on deploy. If the deploy is for prod, then the robots.txt file will be altered to allow indexing.

## How to test
I have tested this out by adding a comment to the robots.txt on dev.

Screen Shot 2021-11-09 at 12 35 31 PM

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-276

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: bin/robot-factory [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+if [[ "$CURRENT_GIT_URL" == "$PROD_GIT_URL" && "$PROD_GIT_BRANCH" == "$CURRENT_GIT_BRANCH" ]];
```
Comment (jasalisbury) [inline bin/robot-factory:3]: Do we want to move this check out into the `circle-ci` config? I _think_ I like it better there, but open to other opinions.

Comment (thewatermethod) [reply bin/robot-factory:3]: Something like this? I'm agnostic. 
Didn't work when I tried it, that's how I got sent down the environment variable route but I think I know why that was.

```yaml
      - when:  # prod
          condition:
            and:
              - equal: [<< pipeline.project.git_url >>, << pipeline.parameters.prod_git_url >>]
              - equal: [<< pipeline.git.branch >>, << pipeline.parameters.prod_git_branch >>]
          steps:
            - build_robot: #build robot
```

Comment (jasalisbury) [reply bin/robot-factory:3]: yeah something like that. If you can't get it to work I think it is fine to leave it in the script.

File: .circleci/config.yml [hunk 2]
```diff
@@ -333,6 +333,15 @@ jobs:
       - run:
           name: Build backend assets
           command: yarn build
+      - when:  # dev
```
Comment (jasalisbury) [inline .circleci/config.yml:336]:
```suggestion
      - when:
```

**Review by jasalisbury — DISMISSED** [review]

**Review by jasalisbury — APPROVED** [review]

**Pull request merged**